core/src/main/java/io/opencensus/stats/StatsManagerFactory.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.stats;
+
+import io.opencensus.common.Clock;
+
+/**
+ * Object that determines which implementation of the {@link StatsManager} to use.
+ */
+public abstract class StatsManagerFactory {
+
+  /**
+   * Returns the default {@link StatsManager}.
+   */
+  public abstract StatsManager getDefaultStatsManager();
+
+  /**
+   * Creates a new {@link StatsManager} for testing.
+   */
+  public abstract StatsManager createTestStatsManager();
```

Comment:
I agree that it would be better to move all testing-related methods out of the API library, but I wasn't sure how to do it.  `createTestStatsManager` is for the integration test use case.  I wanted to provide a way for users to create a new instance of the StatsManager that is exactly like the singleton, so that they can create a new one for each test but know that it has the same behavior as the singleton.

I'm not sure I understand your suggestion.  Do you mean TestingStatsManagerFactory?  How would the class work?  Would it be able to reuse the implementation of the non-test class?  What would users do with the EventQueue?